user/handler.go

```diff
@@ -0,0 +1,70 @@
+package user
+
+import (
+	"github.com/gofiber/fiber/v2"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+type UserHandler struct {
+	Service *UserService
+}
+
+func NewUserHandler(service *UserService) *UserHandler {
+	return &UserHandler{Service: service}
+}
+
+func (h *UserHandler) RegisterRoutes(app *fiber.App) {
+
+	appGroup := app.Group("/api/user")
+
+	appGroup.Post("/", h.CreateUser)
+	appGroup.Get("/:id", h.GetUser)
+	appGroup.Delete("/:id", h.DeleteUser)
+}
+
+func (h *UserHandler) CreateUser(c *fiber.Ctx) error {
+	var createUserDto struct {
+		Nickname string `json:"nickname"`
+		FullName string `json:"fullName"`
+	}
+
+	if err := c.BodyParser(&createUserDto); err != nil {
+		return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{"error": "Invalid request payload"})
+	}
+
+	user, err := h.Service.CreateUser(c.Context(), createUserDto.Nickname, createUserDto.FullName)
```

Comment:
Nickname unique olacağı için eklemeden önce bu nickname zaten var mı diye kontrol et 
Varsa hata fırlat